src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/Runtime/TypeInfos/RuntimeTypeInfo.cs

```diff
@@ -790,10 +790,37 @@ private TypeClassification Classification
         {
             get
             {
+                // We have a very specialized helper to get the base type.
+                // It is not a general purpose base type, but works for the cases we care about.
+                // This avoids bringing in full type resolution support including constructing
+                // generic types.
+                static Type GetLimitedBaseType(RuntimeTypeInfo thisType)
+                {
+                    // If we have a type handle, just use that
```

Comment:
I thought that we got rid of the metadata-only types. Do we still need the metadata-only types? How much benefit do they provide? They feel like an obscure case bug-farm.

Reply:
We got rid of the ones that have type handle, but no metadata. We still have those that have metadata but no type handle. The test has one example where we can come up with such thing. There might be more, I'm not sure I know all situations where this can happen. I think it also might happen if we have a reflected on nested type, but the owning type wasn't touched. We need the metadata. We certainly don't need the vtable.